File: C#/Ethernet/ReadEthernetConfig/ReadEthernetConfig.cs
//-----------------------------------------------------------------------------
// ReadEthernetConfig.cs
//
// Demonstrates how to read the ethernet configuration settings from a LabJack.
//
// support@labjack.com
// Dec. 3, 2013
//-----------------------------------------------------------------------------
using System;
using LabJack;

namespace ReadEthernetConfig
{
    class ReadEthernetConfig
    {
        static void Main(string[] args)
        {
            ReadEthernetConfig rec = new ReadEthernetConfig();
            rec.performActions();
        }

        public void showErrorMessage(LJM.LJMException e)
        {
            Console.Out.WriteLine("LJMException: " + e.ToString());
            Console.Out.WriteLine(e.StackTrace);
        }

        public void performActions()
        {
            int handle = 0;
            int devType = 0;
            int conType = 0;
            int serNum = 0;
            int ipAddr = 0;
            int port = 0;
            int maxBytesPerMB = 0;
            string ipAddrStr = "";

            try
            {
                //Open first found LabJack
                LJM.OpenS("ANY", "ANY", "ANY", ref handle);
                //devType = LJM.CONSTANTS.dtANY; //Any device type
                //conType = LJM.CONSTANTS.ctANY; //Any connection type
                //LJM.Open(devType, conType, "ANY", ref handle);

                LJM.GetHandleInfo(handle, ref devType, ref conType, ref serNum, ref ipAddr, ref port, ref maxBytesPerMB);
                LJM.NumberToIP(ipAddr, ref ipAddrStr);
                Console.WriteLine("Opened a LabJack with Device type: " + devType + ", Connection type: " + conType + ",");
                Console.WriteLine("Serial number: " + serNum + ", IP address: " + ipAddrStr + ", Port: " + port + ",");
                Console.WriteLine("Max bytes per MB: " + maxBytesPerMB);

                //Setup and call eReadNames to read ethernet configuration from
                //the LabJack.
                string[] aNames = new string[] { "ETHERNET_IP",
                    "ETHERNET_SUBNET", "ETHERNET_GATEWAY",
                    "ETHERNET_IP_DEFAULT", "ETHERNET_SUBNET_DEFAULT",
                    "ETHERNET_GATEWAY_DEFAULT", "ETHERNET_DHCP_ENABLE",
                    "ETHERNET_DHCP_ENABLE_DEFAULT" };
                double[] aValues = new double[aNames.Length];
                int numFrames = aNames.Length;
                int errAddr = -1;
                LJM.eReadNames(handle, numFrames, aNames, aValues, ref errAddr);

                Console.WriteLine("\nEthernet configuration: ");
                string str = "";
                for (int i = 0; i < numFrames; i++)
                {
                    if (aNames[i] == "ETHERNET_DHCP_ENABLE")
                    {
                        Console.WriteLine("    " + aNames[i] + " : " + aValues[i]);
                    }
                    else
                    {
                        LJM.NumberToIP(Convert.ToUInt32(aValues[i]), ref str);
                        Console.WriteLine("    " + aNames + " : " + aValues[i] +
                            " - " + str);
                    }
                }
            }
            catch (LJM.LJMException e)
            {
                showErrorMessage(e);
            }

            LJM.CloseAll(); //Close all handles

            Console.WriteLine("\nDone.\nPress the enter key to exit.");
            Console.ReadLine(); // Pause for user
        }
    }
}

File: VB/Ethernet/ReadEthernetConfig/ReadEthernetConfig.vb
﻿'------------------------------------------------------------------------------
' ReadEthernetConfig.vb
'
' Demonstrates how to read the ethernet configuration settings.
'
' support@labjack.com
'------------------------------------------------------------------------------
Option Explicit On

Imports LabJack

Module ReadEthernetConfig

    Sub showErrorMessage(ByVal e As LJM.LJMException)
        Console.WriteLine("LJMException: " & e.ToString)
        Console.WriteLine(e.StackTrace)
    End Sub

    Sub displayHandleInfo(ByVal handle As Integer)
        Dim devType As Integer
        Dim conType As Integer
        Dim serNum As Integer
        Dim ipAddr As Integer
        Dim port As Integer
        Dim maxBytesPerMB As Integer
        Dim ipAddrStr As String = ""

        LJM.GetHandleInfo(handle, devType, conType, serNum, ipAddr, port, _
                          maxBytesPerMB)
        LJM.NumberToIP(ipAddr, ipAddrStr)
        Console.WriteLine("Opened a LabJack with Device type: " & devType & _
                          ", Connection type: " & conType & ",")
        Console.WriteLine("Serial number: " & serNum & ", IP address: " & _
                          ipAddrStr & ", Port: " & port & ",")
        Console.WriteLine("Max bytes per MB: " & maxBytesPerMB)
    End Sub

    Sub Main()
        Dim handle As Integer
        Dim numFrames As Integer
        Dim aNames() As String
        Dim aValues() As Double
        Dim errAddr As Integer = -1
        Dim str As String
        Dim intVal As Integer

        Try
            ' Open first found LabJack
            LJM.OpenS("ANY", "ANY", "ANY", handle)
            'LJM.Open(LJM.CONSTANTS.dtANY, LJM.CONSTANTS.ctANY, "ANY", handle)

            displayHandleInfo(handle)

            'Setup and call eReadNames to read ethernet configuration.
            numFrames = 8
            ReDim aNames(numFrames - 1)
            aNames(0) = "ETHERNET_IP"
            aNames(1) = "ETHERNET_SUBNET"
            aNames(2) = "ETHERNET_GATEWAY"
            aNames(3) = "ETHERNET_IP_DEFAULT"
            aNames(4) = "ETHERNET_SUBNET_DEFAULT"
            aNames(5) = "ETHERNET_GATEWAY_DEFAULT"
            aNames(6) = "ETHERNET_DHCP_ENABLE"
            aNames(7) = "ETHERNET_DHCP_ENABLE_DEFAULT"
            ReDim aValues(numFrames - 1)
            LJM.eReadNames(handle, numFrames, aNames, aValues, errAddr)

            Console.WriteLine("")
            Console.WriteLine("Ethernet configuration: ")
            str = ""
            For i = 0 To numFrames - 1
                If aNames(i).StartsWith("ETHERNET_DHCP_ENABLE") Then
                    Console.WriteLine("    " & aNames(i) & " : " & aValues(i))
                Else
                    intVal = BitConverter.ToInt32(BitConverter.GetBytes(Convert.ToUInt32(aValues(i))), 0)
                    LJM.NumberToIP(intVal, str)
                    Console.WriteLine("    " & aNames(i) & " : " & _
                                      aValues(i) & " - " & str)
                End If
            Next
        Catch ljme As LJM.LJMException
            showErrorMessage(ljme)
        End Try

        LJM.CloseAll()

        Console.WriteLine("")
        Console.WriteLine("Done.")
        Console.WriteLine("Press the enter key to exit.")
        Console.ReadLine() ' Pause for user
    End Sub

End Module
